slimevr-server crashes when OSC used, failing to get libawt

## Describe the bug

When attempting to use any of the OSC features in settings, slimevr-server crashes, this causes a "Connection lost to the server. Trying to reconnect" in the slimevr interface. Here is the slimevr-server output I get during the crash the moment I try to turn on any of the OSC features:


```01:15:54 [INFO] [WebSocketAPI] New connection from: 0:0:0:0:0:0:0:1
01:15:55 [INFO] [WebSocketAPI] New connection from: 0:0:0:0:0:0:0:1
01:16:08 [SEVERE] [WebSocket] Exception on connection 0:0:0:0:0:0:0:1
java.lang.Exception: java.lang.UnsatisfiedLinkError: Can't load library: /nix/store/r0nhkndiz1i7vlxz2j17nhcxnwqlf97r-openjdk-headless-17.0.13+11/lib/openjdk/lib/libawt_xawt.so
        at org.java_websocket.server.WebSocketServer$WebSocketWorker.run(WebSocketServer.java:1113)
Caused by: java.lang.UnsatisfiedLinkError: Can't load library: /nix/store/r0nhkndiz1i7vlxz2j17nhcxnwqlf97r-openjdk-headless-17.0.13+11/lib/openjdk/lib/libawt_xawt.so
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2398)
        at java.base/java.lang.Runtime.load0(Runtime.java:755)
        at java.base/java.lang.System.load(System.java:1970)
        at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
        at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
        at java.base/jdk.internal.loader.NativeLibraries.findFromPaths(NativeLibraries.java:315)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:285)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2403)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
        at java.base/java.lang.System.loadLibrary(System.java:2006)
        at java.desktop/java.awt.Toolkit$2.run(Toolkit.java:1388)
        at java.desktop/java.awt.Toolkit$2.run(Toolkit.java:1386)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
        at java.desktop/java.awt.Toolkit.loadLibraries(Toolkit.java:1385)
        at java.desktop/java.awt.Toolkit.initStatic(Toolkit.java:1423)
        at java.desktop/java.awt.Toolkit.<clinit>(Toolkit.java:1397)
        at java.desktop/java.awt.Color.<clinit>(Color.java:277)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:375)
        at com.illposed.osc.LibraryInfo.hasAwtColor(LibraryInfo.java:302)
        at com.illposed.osc.LibraryInfo.<clinit>(LibraryInfo.java:54)
        at com.illposed.osc.transport.udp.UDPTransport.<init>(UDPTransport.java:65)
        at com.illposed.osc.transport.OSCPort.<init>(OSCPort.java:44)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:112)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:127)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:136)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:149)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:164)
        at com.illposed.osc.transport.OSCPortIn.<init>(OSCPortIn.java:174)
        at dev.slimevr.osc.VRCOSCHandler.updateOscReceiver(VRCOSCHandler.kt:157)
        at dev.slimevr.osc.VRCOSCHandler.refreshSettings(VRCOSCHandler.kt:94)
        at dev.slimevr.protocol.rpc.settings.RPCSettingsHandler.onChangeSettingsRequest(RPCSettingsHandler.kt:140)
        at dev.slimevr.protocol.rpc.settings.RPCSettingsHandler._init_$lambda$1(RPCSettingsHandler.kt:31)
        at dev.slimevr.protocol.rpc.RPCHandler.onMessage(RPCHandler.kt:413)
        at dev.slimevr.protocol.ProtocolAPI.onMessage(ProtocolAPI.java:44)
        at dev.slimevr.websocketapi.WebsocketAPI.onMessage(WebsocketAPI.java:85)
        at org.java_websocket.server.WebSocketServer.onWebsocketMessage(WebSocketServer.java:738)
        at org.java_websocket.drafts.Draft_6455.processFrameBinary(Draft_6455.java:959)
        at org.java_websocket.drafts.Draft_6455.processFrame(Draft_6455.java:912)
        at org.java_websocket.WebSocketImpl.decodeFrames(WebSocketImpl.java:397)
        at org.java_websocket.WebSocketImpl.decode(WebSocketImpl.java:229)
        at org.java_websocket.server.WebSocketServer$WebSocketWorker.doDecode(WebSocketServer.java:1134)
        at org.java_websocket.server.WebSocketServer$WebSocketWorker.run(WebSocketServer.java:1106)
01:16:08 [INFO] [WebSocketAPI] Disconnected: 0:0:0:0:0:0:0:1, (1001) Got error on server side: java.lang.Exception caused by java.lang.UnsatisfiedLinkError. Remote: false 

```

## Steps To Reproduce

Steps to reproduce the behavior:

1. Install slimevr and slimevr-server packages
2. Attempt to turn on any of the three OSC features in slimevr interface.

## Expected behavior

(If I understand correctly) These features cause slimevr-server to create a server for games that use full-body-tracking to receive the animations created by slimevr.

## Screenshots

<!-- If applicable, add screenshots to help explain your problem: -->

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.68, NixOS, 25.05 (Warbler), 25.05pre732674.a27871180d30`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

## Notify maintainers
@ImUrX @gale-username 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sl/slimevr-server/package.nix
+++ b/pkgs/by-name/sl/slimevr-server/package.nix
@@ -4,13 +4,14 @@
   runCommand,
   replaceVars,
   slimevr,
-  jdk17_headless,
+  jdk17,
   gradle,
   hidapi,
   makeWrapper,
 }:
 let
-  java = jdk17_headless;
+  # JDK can't be headless while SlimeVR uses JavaOSC 0.8.
+  java = jdk17;
   # Without this the hidapi bundled with `org.hid4java:hid4java` will be used.
   # The bundled version won't be able to find `libudev.so.1`.
   javaOptions =

